packages/entity/src/utils/EntityPrivacyUtils.ts

```
@@ -0,0 +1,311 @@
+  TMPrivacyPolicy extends EntityPrivacyPolicy<
+    TMFields,
+    TMID,
+    TMViewerContext,
+    TMEntity,
+    TMSelectedFields
+  >,
+  TMSelectedFields extends keyof TMFields = keyof TMFields
+>(
+  entityClass: IEntityClass<
+    TMFields,
+    TMID,
+    TMViewerContext,
+    TMEntity,
+    TMPrivacyPolicy,
+    TMSelectedFields
+  >,
+  sourceEntity: TMEntity,
+  queryContext: EntityQueryContext = sourceEntity
+    .getViewerContext()
+    .getViewerScopedEntityCompanionForClass(entityClass)
+    .getQueryContextProvider()
+    .getQueryContext()
+): Promise<boolean> {
+  return await canViewerUpdateInternalAsync(
+    entityClass,
+    sourceEntity,
+    /* cascadingDeleteCause */ null,
+    queryContext
+  );
+}
+
+async function canViewerUpdateInternalAsync<
+  TMFields extends object,
+  TMID extends NonNullable<TMFields[TMSelectedFields]>,
+  TMViewerContext extends ViewerContext,
+  TMEntity extends Entity<TMFields, TMID, TMViewerContext, TMSelectedFields>,
+  TMPrivacyPolicy extends EntityPrivacyPolicy<
+    TMFields,
+    TMID,
+    TMViewerContext,
+    TMEntity,
+    TMSelectedFields
+  >,
+  TMSelectedFields extends keyof TMFields = keyof TMFields
+>(
+  entityClass: IEntityClass<
+    TMFields,
+    TMID,
+    TMViewerContext,
+    TMEntity,
+    TMPrivacyPolicy,
+    TMSelectedFields
+  >,
+  sourceEntity: TMEntity,
+  cascadingDeleteCause: EntityCascadingDeletionInfo | null,
+  queryContext: EntityQueryContext
+): Promise<boolean> {
+  const companion = sourceEntity
+    .getViewerContext()
+    .getViewerScopedEntityCompanionForClass(entityClass);
+  const privacyPolicy = companion.entityCompanion.privacyPolicy;
+  const evaluationResult = await asyncResult(
+    privacyPolicy.authorizeUpdateAsync(
+      sourceEntity.getViewerContext(),
+      queryContext,
+      { cascadingDeleteCause },
+      sourceEntity,
+      companion.getMetricsAdapter()
+    )
+  );
+  return evaluationResult.ok;
```

Shouldn't we check the error type like we check it for deletes? Specifically, if the database is down or the code is buggy ("undefined is not an object") we should rethrow instead of returning false.